### PR TITLE
Integration tests setup

### PR DESCRIPTION
**Pull request by Tulsishah**
### Description

Making Mounting, Unmounting, TestSetupDir, and LogandExit function public to use in another integration testing. 
Commands to run Integrationtests
cd gcsfuse/tools/integration_tests/implicitdir
1. Bucket
  - GODEBUG=asyncpreemptoff=1 go test -v --testbucket=temp-0011001100-bucket(Bucket Name)
2. Already mounted Bucket
  - GODEBUG=asyncpreemptoff=1 go test -v --mountedDirectory=(Aboslute path of mountDir)

### Testing

1. Manual - Tested manually implicitdir_test (all the tests are passing in both the cases testbucket and mountdir)
4. UnitTest - N/A
5. IntegrationTest - N/A

### CODE REVIEW

File: tools/integration_tests/implicitdir/implicitdir_test.go [hunk 1]
```diff
@@ -166,35 +77,40 @@ func executeTest(m *testing.M) (successCode int) {
...
 	successCode = m.Run()
 
-	os.RemoveAll(mntDir)
+	os.RemoveAll(setup.MntDir())
 
 	return successCode
 }
 
-func executeTestForFlags(flags []string, m *testing.M) (successCode int) {
+func executeTestForFlags(flags [][]string, m *testing.M) (successCode int) {
```
Comment (Tulsishah) [inline tools/integration_tests/implicitdir/implicitdir_test.go:92]: Before we could pass only one flag at a time
Eg.
1. --enable-storage-client-libaray 
Here I converted flags in 2D to pass multiple flags in one tests.
E.g. = 
1. --enable-storage-client-libaray , --implicit-dirs, --o=ro

File: tools/integration_tests/setup/setup.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+		return fmt.Errorf("TempDir: %w\n", err)
+	}
+	SetTestDir(testDir)
+
+	err = util.BuildGcsfuse(TestDir())
+	if err != nil {
+		return fmt.Errorf("BuildGcsfuse(%q): %w\n", TestDir(), err)
+	}
+
+	SetBinFile(path.Join(TestDir(), "bin/gcsfuse"))
```
Comment (vadlakondaswetha) [inline tools/integration_tests/setup/setup.go:87]: setter and getter are for other packages to access these variable. not for this package and more importantly this file

File: tools/integration_tests/implicitdir/implicitdir_test.go [hunk 3]
```diff
@@ -205,32 +121,32 @@ func executeTestForFlags(flags []string, m *testing.M) (successCode int) {
 func TestMain(m *testing.M) {
```
Comment (vadlakondaswetha) [inline tools/integration_tests/implicitdir/implicitdir_test.go:121]: why are these methods still in this file

Comment (Tulsishah) [reply tools/integration_tests/implicitdir/implicitdir_test.go:27]: fixed

File: tools/integration_tests/setup/setup.go (hunk 2 above)
Comment (Tulsishah) [reply tools/integration_tests/setup/setup.go]: fixed

File: tools/integration_tests/setup/setup.go [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+
+		err = UnMount()
+		if err != nil {
+			LogAndExit(fmt.Sprintf("Error in unmounting bucket: %v", err))
+		}
+
+		// Print flag on which test fails
+		if successCode != 0 {
+			var f string
+			for j := 0; j < len(flags[i]); j++ {
```
Comment (sethiay) [inline tools/integration_tests/setup/setup.go:230]: Can we use strings.Join here?

**Review by sethiay — COMMENTED** [review]
can you also write the commands in Description of this PR to run these tests using mountedDirectory and testBucket flags ? (write the commands assuming that you are in gcsfuse directory)

**Pull request merged**